core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -744,7 +747,12 @@ constructor(
                 MediaStoreOutputOptions.Builder(
                     application.contentResolver,
                     MediaStore.Video.Media.EXTERNAL_CONTENT_URI
-                )
+                ).apply {
+                    // apply duration limit if applicable
+                    if (currentSettings.value?.maxVideoDurationMillis != -1L) {
+                        setDurationLimitMillis(currentSettings.value!!.maxVideoDurationMillis)
```

Comment:
Instead of reading from `currentSettings` here, let's move the `maxDurationMillis` to be an argument to this function. It should be sent as a property in the `VideoCaptureControlEvent.StartVideoRecording` event so that it is closer associated with calling `CameraUseCase.startVideoRecording`.

Reply:
I am a bit confused why it would need to be in a `VideoCaptureControlEvent.StartVideoRecording` event  when `transientSettings` is available as in `RunVideoRecording()`

Reply:
It's a matter of organizing concepts and keeping things more deterministic. 

Since events are queued in the `videoCaptureControlEvents` Channel, if we put the duration limit in `TransientSessionSettings`, it's possible that the duration limit could change in the time between sending the `StartVideoRecording` event and when the event is actually serviced, making it less deterministic.

When you start a recording, you should know what the duration limit will be. By putting the duration limit in the event, when you read the code, you can clearly see "this event will start a recording with a limit of 5 seconds" rather than "this event will start a recording that may or may not be limited in duration".  This will reduce cognitive load of those who need to maintain code. They won't have to go searching for what the duration will be in the code.